### ISSUE TITLE
CI for the backend

### ISSUE THREAD

**Issue by smarter** [lampepfl/dotty#2295]
When a PR is made against the backend repo (https://github.com/lampepfl/scala/tree/sharing-backend), it should automatically run the dotty CI with the backend submodule set to the PR's latest commit.

**Comment by liufengyun** [comment]
No longer relevant after #5869.
